lib/rules/no-empty-static-block.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * @fileoverview Rule to disallow empty static blocks.
+ * @author Sosuke Suzuki
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const astUtils = require("./utils/ast-utils");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('../shared/types').Rule} */
+module.exports = {
+    meta: {
+        type: "suggestion",
+
+        docs: {
+            description: "Disallows empty static blocks",
+            recommended: false,
+            url: "https://eslint.org/docs/rules/no-empty-static-block"
+        },
+
+        schema: [],
+
+        messages: {
+            unexpected: "Unexpected empty static block."
+        }
+    },
+
+    create(context) {
+        const sourceCode = context.getSourceCode();
+
+        return {
+            StaticBlock(node) {
+                const innerComments = sourceCode.getTokens(node, {
+                    includeComments: true,
+                    filter: astUtils.isCommentToken
+                });
```

Comment:
I think that only comments inside `{ }` should count.

```js
class Foo {
  static // this comment should not suppress the error
  {

  }
}
```

Reply:
What do you think about [c697d95](https://github.com/eslint/eslint/pull/16325/commits/c697d952001eb1de7a0ffcbb00ee8f420582d275) ?